packages/hub/node-tests/routes/push-notification-registrations-test.ts

```diff
@@ -0,0 +1,134 @@
+      .send({})
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(401)
+      .expect({
+        errors: [
+          {
+            status: '401',
+            title: 'No valid auth token',
+          },
+        ],
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+  });
+
+  it('persists push notification registration', async function () {
+    let payload = {
+      data: {
+        type: 'push-notification-registration',
+        attributes: {
+          'push-client-id': 'FIREBASE_USER_ID',
+        },
+      },
+    };
+
+    await request()
+      .post('/api/push-notification-registrations')
+      .send(payload)
+      .set('Authorization', 'Bearer abc123--def456--ghi789')
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(201)
+      .expect({
+        data: {
+          type: 'push-notification-registration',
+          attributes: {
+            'owner-address': stubUserAddress,
+            'push-client-id': 'FIREBASE_USER_ID',
+            'disabled-at': null,
+          },
+        },
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+  });
+});
+
+describe('DELETE /api/push-notification-registrations', function () {
+  this.beforeEach(function () {
+    registry(this).register('authentication-utils', StubAuthenticationUtils);
+  });
+  let { request, getContainer } = setupHub(this);
+
+  it('returns 401 without bearer token', async function () {
+    await request()
+      .post('/api/push-notification-registrations')
+      .send({})
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(401)
+      .expect({
+        errors: [
+          {
+            status: '401',
+            title: 'No valid auth token',
+          },
+        ],
+      })
+      .expect('Content-Type', 'application/vnd.api+json');
+  });
+
+  it('deletes push notification registration', async function () {
+    let pushNotificationRegistrationQueries = await getContainer().lookup('push-notification-registration-queries');
+
+    await pushNotificationRegistrationQueries.insert({
+      ownerAddress: stubUserAddress,
+      pushClientId: 'FIREBASE_USER_ID',
+      disabledAt: null,
+    });
+
+    await request()
+      .delete('/api/push-notification-registrations')
+      .send({})
+      .set('Authorization', 'Bearer abc123--def456--ghi789')
+      .set('Accept', 'application/vnd.api+json')
+      .set('Content-Type', 'application/vnd.api+json')
+      .expect(200);
+
+    let dbManager = await getContainer().lookup('database-manager');
+    let db = await dbManager.getClient();
+
+    let queryResult = await db.query('SELECT * FROM push_notification_registrations WHERE owner_address = $1', [
+      stubUserAddress,
+    ]);
+
+    expect(queryResult.rows.length).to.equal(0);
+  });
+});
```

Comment:
A spec to describe what happens if a registration already exists for this device/EOA pair should be added here. I would suggest this request should silently succeed and perhaps clear the disabled_at field if it has been set.

Reply:
Thanks for the suggestion. I added that case.